File: Olympiad/COCI/official/2007/contest4_solutions/sibice.cpp

/*
  Croatian Open Competition in Informatics 2006/2007
  Contest 4 - Task SIBICE
  Programming language C++
*/

#include <cstdio>

int main(void) {
   int n, w, h, k;
   scanf( "%d%d%d", &n, &w, &h );
   for( int i = 0; i < n; i++ ) {
      scanf( "%d", &k );
      printf( w*w+h*h >= k*k ? "DA\n": "NE\n" );
   }
   return 0;
}

File: Olympiad/COCI/official/2007/contest4_solutions/sibice.pas

(*
  Croatian Open Competition in Informatics 2006/2007
  Contest 4 - Task SIBICE
  Programming language Pascal
*)

program sibice;

var i, j, n, w, h: integer;

begin
    readln( n, w, h );
    for i := 1 to n do begin
        readln( j );
        if( j*j <= w*w + h*h ) then writeln( 'DA' )
        else writeln( 'NE' );
    end;
end.
